app-backend/api/src/main/scala/uploads/Routes.scala

```diff
@@ -62,9 +63,25 @@ trait UploadRoutes extends Authentication
   }
 
   def createUpload: Route = authenticate { user =>
+    implicit def stringAsJavaURI(uri: String): URI = new URI(uri)
     entity(as[Upload.Create]) { newUpload =>
       authorize(user.isInRootOrSameOrganizationAs(newUpload)) {
-        onSuccess(write[Upload](Uploads.insertUpload(newUpload, user))) { upload =>
+        val uploadToInsert = (newUpload.uploadType.toString, newUpload.source) match {
+          case ("S3", Some(source)) => {
```

Comment:
It's better to mach here on UploadType itself and not on its string representation.

```scala
(newUpload.uploadType, newUpload.source) match {
  case (UploadType.S3, Some(source)) => ...
}
```